federatedscope/attack/auxiliary/poisoning_data.py

```diff
@@ -0,0 +1,302 @@
+import torch
+from PIL import Image
+import numpy as np
+from torchvision.datasets import MNIST, EMNIST, CIFAR10
+from torchvision.datasets import DatasetFolder
+from torchvision import transforms
+from federatedscope.core.auxiliaries.transform_builder import get_transform
+from federatedscope.attack.auxiliary.backdoor_utils import selectTrigger
+from torch.utils.data import DataLoader, Dataset
+from federatedscope.attack.auxiliary.backdoor_utils import normalize
+import matplotlib
+import pickle
+import logging
+import os
+
+logger = logging.getLogger(__name__)
+
+
+def load_poisoned_dataset_edgeset(data, ctx, mode):
+
+    transforms_funcs = get_transform(ctx, 'torchvision')['transform']
+    load_path = ctx.attack.edge_path
```

Comment:
Since the edge_path defaults to `''`, the poison data will be saved locally. 
Please save them in the directory `data.root`.